extension/parquet/parquet-extension.cpp

```diff
@@ -73,6 +75,56 @@ class ParquetScanFunction : public TableFunction {
 		return move(result);
 	}
 
+	static unique_ptr<BaseStatistics> parquet_scan_stats(ClientContext &context, const FunctionData *bind_data_,
+	                                                     column_t column_index) {
+		auto &bind_data = (ParquetReadBindData &)*bind_data_;
+
+		if (column_index == COLUMN_IDENTIFIER_ROW_ID) {
+			return nullptr;
+		}
+
+		// we do not want to parse the Parquet metadata for the sole purpose of getting column statistics
+
+		// We already parsed the metadata for the first file in a glob because we need some type info.
+		auto overall_stats =
+		    ParquetReader::ReadStatistics(bind_data.initial_reader->return_types[column_index], column_index,
+		                                  bind_data.initial_reader->metadata->metadata.get());
+
+		if (!overall_stats) {
+			return nullptr;
+		}
+
+		// if there is only one file in the glob (quite common case), we are done
+		if (bind_data.files.size() < 2) {
+			return overall_stats;
+		} else if (context.db.config.object_cache_enable) {
+			// for more than one file, we could be lucky and metadata for *every* file is in the object cache (if
+			// enabled at all)
+			FileSystem &fs = FileSystem::GetFileSystem(context);
+			for (idx_t file_idx = 1; file_idx < bind_data.files.size(); file_idx++) {
+				auto &file_name = bind_data.files[file_idx];
+				auto metadata = dynamic_pointer_cast<ParquetFileMetadataCache>(context.db.object_cache->Get(file_name));
+				auto handle = fs.OpenFile(file_name, FileFlags::FILE_FLAGS_READ);
+				// but we need to check if the metadata cache entries are current
+				if (!metadata || (fs.GetLastModifiedTime(*handle) >= metadata->read_time)) {
```

Comment:
This seems like a copy of the metadata extract from the parquet reader, perhaps better to move it to a function?

Reply:
Its only the check part, not sure it warrants unification